packages/lib-classifier/src/store/UserProjectPreferencesStore.js

```diff
@@ -89,8 +92,30 @@ const UserProjectPreferencesStore = types
       }
     }
 
-    function updateUPP (changes) {
-      console.log('TODO: UPP PUT request', changes)
+    function * updateUPP (changes) {
+      const { authClient } = getRoot(self)
+      const upp = self.resources.get(self.active.id)
+      if (upp) {
+        const authorization = yield getBearerToken(authClient)
+        const { type } = self
+        const client = getRoot(self).client.panoptes
+        self.loadingState = asyncStates.putting
+        try {
+          const newUPP = merge({}, upp, changes)
+          const headers = {
+            authorization,
+            etag: self.headers.etag
+          }
+          if (self.headers['Last-Modified']) headers.lastModified = self.headers['Last-Modified']
+          const response = yield client.put(`/${type}/${upp.id}`, { [type]: { preferences: newUPP.preferences } }, headers)
+          const updatedUPP = response.body[type][0]
+          self.setUPP(updatedUPP)
+          self.loadingState = asyncStates.success
```

Comment:
The ETag and Last-Modified time have likely changed, so set those here for the updated resource?

Reply:
Noting for future work: this works for a single resource but is going to get hairy when a model has to manage multiple resources, each with their own ETag and Last-Modified time.

Reply:
I don't think MST models typically should should manage multiple resources. I think we could decide architecturally if the model is going to need to call updates to the API, that that resource should have its own model.

Reply:
True. I was thinking of a workflow's tutorial and minicourse, but they can each have their own model.

Reply:
Yeah, there's some murkiness in specific cases. But in the classifier, we don't need to PUT that resource, so we can have a single tutorial model store that stores its attached media along side it for convenience. I think when we get around to finishing the lab, we could think about more strictly having separate models for each resource (also whether or not we want to stick with redux with that app or not).